[Automation] java run fails for org.jboss.narayana.jta:jta:7.3.1.Final

## java run on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: java run
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="7.3.1.Final" ./gradlew clean javaTest -Pcoordinates="org.jboss.narayana.jta:jta:5.12.4.Final"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/24617275190/job/71981567117
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
hestrator.execute(EngineExecutionOrchestrator.java:127)
                at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:90)
                at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:55)
                at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:102)
                at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:54)
                at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)
                at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
                at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
                at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.processAllTestClasses(JUnitPlatformTestClassProcessor.java:135)
                at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.access$000(JUnitPlatformTestClassProcessor.java:110)
                at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor.stop(JUnitPlatformTestClassProcessor.java:104)
                at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.stop(SuiteTestClassProcessor.java:64)
                at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
                at java.base/java.lang.reflect.Method.invoke(Method.java:580)
                at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
                at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
                at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
                at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:92)
                at jdk.proxy2/jdk.proxy2.$Proxy6.stop(Unknown Source)
                at org.gradle.api.internal.tasks.testing.worker.TestWorker$3.run(TestWorker.java:194)
                at org.gradle.api.internal.tasks.testing.worker.TestWorker.executeAndMaintainThreadName(TestWorker.java:126)
                at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:103)
                at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:63)
                at org.gradle.process.internal.worker.child.ActionExecutionWorker.execute(ActionExecutionWorker.java:56)
                at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:122)
                at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:72)
                at worker.org.gradle.process.internal.worker.GradleWorkerMain.run(GradleWorkerMain.java:69)
                at worker.org.gradle.process.internal.worker.GradleWorkerMain.main(GradleWorkerMain.java:74)

XAResourceRecordTest > saveAndRestoreStateWithSerializableXaResource() FAILED
    java.lang.ExceptionInInitializerError
        at org_jboss_narayana_jta.jta.XAResourceRecordTest.saveAndRestoreStateWithSerializableXaResource(XAResourceRecordTest.java:28)

        Caused by:
        java.lang.RuntimeException: java.lang.NoClassDefFoundError: Could not initialize class com.arjuna.common.logging.commonLogger
            at com.arjuna.common.internal.util.propertyservice.BeanPopulator.getNamedInstance(BeanPopulator.java:75)
            at com.arjuna.common.internal.util.propertyservice.BeanPopulator.getDefaultInstance(BeanPopulator.java:38)
            at com.arjuna.ats.arjuna.common.arjPropertyManager.getCoordinatorEnvironmentBean(arjPropertyManager.java:36)
            at com.arjuna.ats.arjuna.coordinator.AbstractRecord.<clinit>(AbstractRecord.java:691)
            ... 1 more

            Caused by:
            java.lang.NoClassDefFoundError: Could not initialize class com.arjuna.common.logging.commonLogger
                at com.arjuna.common.util.propertyservice.AbstractPropertiesFactory.getPropertiesFromFile(AbstractPropertiesFactory.java:64)
                at com.arjuna.common.util.propertyservice.AbstractPropertiesFactory.initDefaultProperties(AbstractPropertiesFactory.java:172)
                at com.arjuna.common.util.propertyservice.AbstractPropertiesFactory.getDefaultProperties(AbstractPropertiesFactory.java:38)
                at com.arjuna.common.util.propertyservice.PropertiesFactory.getDefaultProperties(PropertiesFactory.java:32)
                at com.arjuna.common.internal.util.propertyservice.BeanPopulator.getNamedInstance(BeanPopulator.java:71)
                ... 4 more

                Caused by:
                java.lang.ExceptionInInitializerError: Exception java.lang.IllegalArgumentException: Invalid logger interface com.arjuna.common.logging.commonI18NLogger (implementation not found) [in thread "Test worker"]
                    at org.jboss.logging.Logger.doGetMessageLogger(Logger.java:2630)
                    at org.jboss.logging.Logger.getMessageLogger(Logger.java:2536)
                    at org.jboss.logging.Logger.getMessageLogger(Logger.java:2521)
                    at com.arjuna.common.logging.commonLogger.<clinit>(commonLogger.java:16)
                    at com.arjuna.common.util.propertyservice.AbstractPropertiesFactory.getPropertiesFromFile(AbstractPropertiesFactory.java:64)
                    at com.arjuna.common.util.propertyservice.AbstractPropertiesFactory.initDefaultProperties(AbstractPropertiesFactory.java:172)
                    at com.arjuna.common.util.propertyservice.AbstractPropertiesFactory.getDefaultProperties(AbstractPropertiesFactory.java:38)
                    at com.arjuna.common.util.propertyservice.PropertiesFactory.getDefaultProperties(PropertiesFactory.java:32)
                    at com.arjuna.common.internal.util.propertyservice.BeanPopulator.getNamedInstance(BeanPopulator.java:71)
                    at com.arjuna.common.internal.util.propertyservice.BeanPopulator.getDefaultInstance(BeanPopulator.java:38)
                    at com.arjuna.ats.jta.common.jtaPropertyManager.getJTAEnvironmentBean(jtaPropertyManager.java:18)
                    at org_jboss_narayana_jta.jta.JNDIManagerTest.bindTransactionSynchronizationRegistryImplementationUsesConfiguredClass(JNDIManagerTest.java:29)
                    at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
                    at java.base/java.lang.reflect.Method.invoke(Method.java:580)
                    at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
                    at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
                    at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
                    at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:156)
                    at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:147)
                    at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:86)
                    at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(InterceptingExecutableInvoker.java:103)
                    at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.lambda$invoke$0(InterceptingExecutableInvoker.java:93)
                    at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
                    at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
                    at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
                    at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
                    at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:92)
                    at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:86)
                    at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$7(TestMethodTestDescriptor.java:217)
                    at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
                    at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:213)
                    at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:138)
                    at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:68)
                    at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:151)
                    at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
                    at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
                    at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
                    at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
                    at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
                    at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
                    at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
                    at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
                    at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
                    at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
                    at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
                    at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
                    at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
                    at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
                    at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
                    at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
                    at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
                    at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
                    at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
                    at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
                    at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
                    at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
                    at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
                    at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
                    at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
                    at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
                    at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
                    at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35)
                    at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
                    at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:54)
                    at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:147)
                    at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:127)
                    at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:90)
                    at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:55)
                    at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:102)
                    at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:54)
                    at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)
                    at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
                    at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
                    at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.processAllTestClasses(JUnitPlatformTestClassProcessor.java:135)
                    at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.access$000(JUnitPlatformTestClassProcessor.java:110)
                    at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor.stop(JUnitPlatformTestClassProcessor.java:104)
                    at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.stop(SuiteTestClassProcessor.java:64)
                    at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
                    at java.base/java.lang.reflect.Method.invoke(Method.java:580)
                    at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
                    at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
                    at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
                    at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:92)
                    at jdk.proxy2/jdk.proxy2.$Proxy6.stop(Unknown Source)
                    at org.gradle.api.internal.tasks.testing.worker.TestWorker$3.run(TestWorker.java:194)
                    at org.gradle.api.internal.tasks.testing.worker.TestWorker.executeAndMaintainThreadName(TestWorker.java:126)
                    at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:103)
                    at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:63)
                    at org.gradle.process.internal.worker.child.ActionExecutionWorker.execute(ActionExecutionWorker.java:56)
                    at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:122)
                    at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:72)
                    at worker.org.gradle.process.internal.worker.GradleWorkerMain.run(GradleWorkerMain.java:69)
                    at worker.org.gradle.process.internal.worker.GradleWorkerMain.main(GradleWorkerMain.java:74)

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss.narayana.jta/jta/5.12.4.Final/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.


3 tests completed, 3 failed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss.narayana.jta/jta/5.12.4.Final/build/reports/tests/test/index.html

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 2s
For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
11 actionable tasks: 3 executed, 8 up-to-date

> Task :javaTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':javaTest'.
> Java tests failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 4s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'java run' for 7.3.1.Final failed with exit code 1.
FAILED[java run][7.3.1.Final][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="7.3.1.Final" ./gradlew clean javaTest -Pcoordinates="org.jboss.narayana.jta:jta:5.12.4.Final"]
```

## java run on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: java run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="7.3.1.Final" ./gradlew clean javaTest -Pcoordinates="org.jboss.narayana.jta:jta:5.12.4.Final"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/24617275190/job/71981567097
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
estrator.execute(EngineExecutionOrchestrator.java:127)
                at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:90)
                at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:55)
                at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:102)
                at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:54)
                at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)
                at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
                at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
                at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.processAllTestClasses(JUnitPlatformTestClassProcessor.java:135)
                at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.access$000(JUnitPlatformTestClassProcessor.java:110)
                at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor.stop(JUnitPlatformTestClassProcessor.java:104)
                at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.stop(SuiteTestClassProcessor.java:64)
                at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
                at java.base/java.lang.reflect.Method.invoke(Method.java:580)
                at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
                at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
                at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
                at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:92)
                at jdk.proxy2/jdk.proxy2.$Proxy6.stop(Unknown Source)
                at org.gradle.api.internal.tasks.testing.worker.TestWorker$3.run(TestWorker.java:194)
                at org.gradle.api.internal.tasks.testing.worker.TestWorker.executeAndMaintainThreadName(TestWorker.java:126)
                at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:103)
                at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:63)
                at org.gradle.process.internal.worker.child.ActionExecutionWorker.execute(ActionExecutionWorker.java:56)
                at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:122)
                at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:72)
                at worker.org.gradle.process.internal.worker.GradleWorkerMain.run(GradleWorkerMain.java:69)
                at worker.org.gradle.process.internal.worker.GradleWorkerMain.main(GradleWorkerMain.java:74)

XAResourceRecordTest > saveAndRestoreStateWithSerializableXaResource() FAILED
    java.lang.ExceptionInInitializerError
        at org_jboss_narayana_jta.jta.XAResourceRecordTest.saveAndRestoreStateWithSerializableXaResource(XAResourceRecordTest.java:28)

        Caused by:
        java.lang.RuntimeException: java.lang.NoClassDefFoundError: Could not initialize class com.arjuna.common.logging.commonLogger
            at com.arjuna.common.internal.util.propertyservice.BeanPopulator.getNamedInstance(BeanPopulator.java:75)
            at com.arjuna.common.internal.util.propertyservice.BeanPopulator.getDefaultInstance(BeanPopulator.java:38)
            at com.arjuna.ats.arjuna.common.arjPropertyManager.getCoordinatorEnvironmentBean(arjPropertyManager.java:36)
            at com.arjuna.ats.arjuna.coordinator.AbstractRecord.<clinit>(AbstractRecord.java:691)
            ... 1 more

            Caused by:
            java.lang.NoClassDefFoundError: Could not initialize class com.arjuna.common.logging.commonLogger
                at com.arjuna.common.util.propertyservice.AbstractPropertiesFactory.getPropertiesFromFile(AbstractPropertiesFactory.java:64)
                at com.arjuna.common.util.propertyservice.AbstractPropertiesFactory.initDefaultProperties(AbstractPropertiesFactory.java:172)
                at com.arjuna.common.util.propertyservice.AbstractPropertiesFactory.getDefaultProperties(AbstractPropertiesFactory.java:38)
                at com.arjuna.common.util.propertyservice.PropertiesFactory.getDefaultProperties(PropertiesFactory.java:32)
                at com.arjuna.common.internal.util.propertyservice.BeanPopulator.getNamedInstance(BeanPopulator.java:71)
                ... 4 more

                Caused by:
                java.lang.ExceptionInInitializerError: Exception java.lang.IllegalArgumentException: Invalid logger interface com.arjuna.common.logging.commonI18NLogger (implementation not found) [in thread "Test worker"]
                    at org.jboss.logging.Logger.doGetMessageLogger(Logger.java:2630)
                    at org.jboss.logging.Logger.getMessageLogger(Logger.java:2536)
                    at org.jboss.logging.Logger.getMessageLogger(Logger.java:2521)
                    at com.arjuna.common.logging.commonLogger.<clinit>(commonLogger.java:16)
                    at com.arjuna.common.util.propertyservice.AbstractPropertiesFactory.getPropertiesFromFile(AbstractPropertiesFactory.java:64)
                    at com.arjuna.common.util.propertyservice.AbstractPropertiesFactory.initDefaultProperties(AbstractPropertiesFactory.java:172)
                    at com.arjuna.common.util.propertyservice.AbstractPropertiesFactory.getDefaultProperties(AbstractPropertiesFactory.java:38)
                    at com.arjuna.common.util.propertyservice.PropertiesFactory.getDefaultProperties(PropertiesFactory.java:32)
                    at com.arjuna.common.internal.util.propertyservice.BeanPopulator.getNamedInstance(BeanPopulator.java:71)
                    at com.arjuna.common.internal.util.propertyservice.BeanPopulator.getDefaultInstance(BeanPopulator.java:38)
                    at com.arjuna.ats.jta.common.jtaPropertyManager.getJTAEnvironmentBean(jtaPropertyManager.java:18)
                    at org_jboss_narayana_jta.jta.JNDIManagerTest.bindTransactionSynchronizationRegistryImplementationUsesConfiguredClass(JNDIManagerTest.java:29)
                    at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
                    at java.base/java.lang.reflect.Method.invoke(Method.java:580)
                    at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
                    at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
                    at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
                    at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:156)
                    at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:147)
                    at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:86)
                    at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(InterceptingExecutableInvoker.java:103)
                    at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.lambda$invoke$0(InterceptingExecutableInvoker.java:93)
                    at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
                    at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
                    at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
                    at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
                    at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:92)
                    at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:86)
                    at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$7(TestMethodTestDescriptor.java:217)
                    at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
                    at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:213)
                    at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:138)
                    at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:68)
                    at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:151)
                    at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
                    at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
                    at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
                    at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
                    at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
                    at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
                    at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
                    at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
                    at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
                    at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
                    at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
                    at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
                    at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
                    at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
                    at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
                    at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
                    at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
                    at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
                    at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
                    at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
                    at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
                    at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
                    at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
                    at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
                    at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
                    at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
                    at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
                    at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35)
                    at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
                    at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:54)
                    at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:147)
                    at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:127)

3 tests completed, 3 failed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss.narayana.jta/jta/5.12.4.Final/build/reports/tests/test/index.html

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 2s
                    at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:90)
                    at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:55)
                    at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:102)
                    at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:54)
                    at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)
                    at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
                    at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
                    at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.processAllTestClasses(JUnitPlatformTestClassProcessor.java:135)
                    at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.access$000(JUnitPlatformTestClassProcessor.java:110)
                    at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor.stop(JUnitPlatformTestClassProcessor.java:104)
                    at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.stop(SuiteTestClassProcessor.java:64)
                    at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
                    at java.base/java.lang.reflect.Method.invoke(Method.java:580)
                    at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
                    at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
                    at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
                    at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:92)
                    at jdk.proxy2/jdk.proxy2.$Proxy6.stop(Unknown Source)
                    at org.gradle.api.internal.tasks.testing.worker.TestWorker$3.run(TestWorker.java:194)
                    at org.gradle.api.internal.tasks.testing.worker.TestWorker.executeAndMaintainThreadName(TestWorker.java:126)
                    at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:103)
                    at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:63)
                    at org.gradle.process.internal.worker.child.ActionExecutionWorker.execute(ActionExecutionWorker.java:56)
                    at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:122)
                    at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:72)
                    at worker.org.gradle.process.internal.worker.GradleWorkerMain.run(GradleWorkerMain.java:69)
                    at worker.org.gradle.process.internal.worker.GradleWorkerMain.main(GradleWorkerMain.java:74)

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss.narayana.jta/jta/5.12.4.Final/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
11 actionable tasks: 3 executed, 8 up-to-date

> Task :javaTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':javaTest'.
> Java tests failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 3s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'java run' for 7.3.1.Final failed with exit code 1.
FAILED[java run][7.3.1.Final][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="7.3.1.Final" ./gradlew clean javaTest -Pcoordinates="org.jboss.narayana.jta:jta:5.12.4.Final"]
```

## java run on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: java run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="7.3.1.Final" ./gradlew clean javaTest -Pcoordinates="org.jboss.narayana.jta:jta:5.12.4.Final"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/24617275190/job/71981567084
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
eExecutionOrchestrator.java:147)
                at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:127)
                at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:90)
                at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:55)
                at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:102)
                at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:54)
                at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)
                at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
                at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
                at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.processAllTestClasses(JUnitPlatformTestClassProcessor.java:135)
                at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.access$000(JUnitPlatformTestClassProcessor.java:110)
                at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor.stop(JUnitPlatformTestClassProcessor.java:104)
                at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.stop(SuiteTestClassProcessor.java:64)
                at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
                at java.base/java.lang.reflect.Method.invoke(Method.java:580)
                at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
                at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
                at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
                at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:92)
                at jdk.proxy2/jdk.proxy2.$Proxy6.stop(Unknown Source)
                at org.gradle.api.internal.tasks.testing.worker.TestWorker$3.run(TestWorker.java:194)
                at org.gradle.api.internal.tasks.testing.worker.TestWorker.executeAndMaintainThreadName(TestWorker.java:126)
                at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:103)
                at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:63)
                at org.gradle.process.internal.worker.child.ActionExecutionWorker.execute(ActionExecutionWorker.java:56)
                at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:122)
                at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:72)
                at worker.org.gradle.process.internal.worker.GradleWorkerMain.run(GradleWorkerMain.java:69)
                at worker.org.gradle.process.internal.worker.GradleWorkerMain.main(GradleWorkerMain.java:74)

XAResourceRecordTest > saveAndRestoreStateWithSerializableXaResource() FAILED
    java.lang.ExceptionInInitializerError
        at org_jboss_narayana_jta.jta.XAResourceRecordTest.saveAndRestoreStateWithSerializableXaResource(XAResourceRecordTest.java:28)

        Caused by:
        java.lang.RuntimeException: java.lang.NoClassDefFoundError: Could not initialize class com.arjuna.common.logging.commonLogger
            at com.arjuna.common.internal.util.propertyservice.BeanPopulator.getNamedInstance(BeanPopulator.java:75)
            at com.arjuna.common.internal.util.propertyservice.BeanPopulator.getDefaultInstance(BeanPopulator.java:38)
            at com.arjuna.ats.arjuna.common.arjPropertyManager.getCoordinatorEnvironmentBean(arjPropertyManager.java:36)
            at com.arjuna.ats.arjuna.coordinator.AbstractRecord.<clinit>(AbstractRecord.java:691)
            ... 1 more

            Caused by:
            java.lang.NoClassDefFoundError: Could not initialize class com.arjuna.common.logging.commonLogger
                at com.arjuna.common.util.propertyservice.AbstractPropertiesFactory.getPropertiesFromFile(AbstractPropertiesFactory.java:64)
                at com.arjuna.common.util.propertyservice.AbstractPropertiesFactory.initDefaultProperties(AbstractPropertiesFactory.java:172)
                at com.arjuna.common.util.propertyservice.AbstractPropertiesFactory.getDefaultProperties(AbstractPropertiesFactory.java:38)
                at com.arjuna.common.util.propertyservice.PropertiesFactory.getDefaultProperties(PropertiesFactory.java:32)
                at com.arjuna.common.internal.util.propertyservice.BeanPopulator.getNamedInstance(BeanPopulator.java:71)
                ... 4 more

                Caused by:
                java.lang.ExceptionInInitializerError: Exception java.lang.IllegalArgumentException: Invalid logger interface com.arjuna.common.logging.commonI18NLogger (implementation not found) [in thread "Test worker"]
                    at org.jboss.logging.Logger.doGetMessageLogger(Logger.java:2630)
                    at org.jboss.logging.Logger.getMessageLogger(Logger.java:2536)
                    at org.jboss.logging.Logger.getMessageLogger(Logger.java:2521)
                    at com.arjuna.common.logging.commonLogger.<clinit>(commonLogger.java:16)
                    at com.arjuna.common.util.propertyservice.AbstractPropertiesFactory.getPropertiesFromFile(AbstractPropertiesFactory.java:64)
                    at com.arjuna.common.util.propertyservice.AbstractPropertiesFactory.initDefaultProperties(AbstractPropertiesFactory.java:172)
                    at com.arjuna.common.util.propertyservice.AbstractPropertiesFactory.getDefaultProperties(AbstractPropertiesFactory.java:38)
                    at com.arjuna.common.util.propertyservice.PropertiesFactory.getDefaultProperties(PropertiesFactory.java:32)
                    at com.arjuna.common.internal.util.propertyservice.BeanPopulator.getNamedInstance(BeanPopulator.java:71)
                    at com.arjuna.common.internal.util.propertyservice.BeanPopulator.getDefaultInstance(BeanPopulator.java:38)
                    at com.arjuna.ats.jta.common.jtaPropertyManager.getJTAEnvironmentBean(jtaPropertyManager.java:18)
                    at org_jboss_narayana_jta.jta.JNDIManagerTest.bindTransactionSynchronizationRegistryImplementationUsesConfiguredClass(JNDIManagerTest.java:29)
                    at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
                    at java.base/java.lang.reflect.Method.invoke(Method.java:580)
                    at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
                    at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
                    at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
                    at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:156)
                    at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:147)
                    at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:86)
                    at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(InterceptingExecutableInvoker.java:103)
                    at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.lambda$invoke$0(InterceptingExecutableInvoker.java:93)
                    at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
                    at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
                    at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
                    at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
                    at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:92)
                    at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:86)
                    at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$7(TestMethodTestDescriptor.java:217)
                    at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
                    at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:213)
                    at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:138)
                    at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:68)
                    at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:151)
                    at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
                    at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
                    at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
                    at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
                    at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
                    at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
                    at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
                    at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
                    at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
                    at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
                    at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
                    at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
                    at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
                    at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
                    at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
                    at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
                    at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
                    at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
                    at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
                    at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
                    at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
                    at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
                    at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
                    at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
                    at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
                    at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
                    at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
                    at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35)
                    at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
                    at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:54)
                    at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:147)
                    at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:127)
                    at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:90)
                    at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:55)
                    at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:102)
                    at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:54)
                    at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)
                    at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
                    at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
                    at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.processAllTestClasses(JUnitPlatformTestClassProcessor.java:135)
                    at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.access$000(JUnitPlatformTestClassProcessor.java:110)
                    at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor.stop(JUnitPlatformTestClassProcessor.java:104)
                    at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.stop(SuiteTestClassProcessor.java:64)
                    at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
                    at java.base/java.lang.reflect.Method.invoke(Method.java:580)
                    at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
                    at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
                    at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
                    at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:92)
                    at jdk.proxy2/jdk.proxy2.$Proxy6.stop(Unknown Source)
                    at org.gradle.api.internal.tasks.testing.worker.TestWorker$3.run(TestWorker.java:194)
                    at org.gradle.api.internal.tasks.testing.worker.TestWorker.executeAndMaintainThreadName(TestWorker.java:126)
                    at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:103)
                    at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:63)
                    at org.gradle.process.internal.worker.child.ActionExecutionWorker.execute(ActionExecutionWorker.java:56)
                    at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:122)
                    at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:72)
                    at worker.org.gradle.process.internal.worker.GradleWorkerMain.run(GradleWorkerMain.java:69)
                    at worker.org.gradle.process.internal.worker.GradleWorkerMain.main(GradleWorkerMain.java:74)


3 tests completed, 3 failed
[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss.narayana.jta/jta/5.12.4.Final/build/reports/problems/problems-report.html


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss.narayana.jta/jta/5.12.4.Final/build/reports/tests/test/index.html

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).
Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1s
11 actionable tasks: 3 executed, 8 up-to-date

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':javaTest'.
> Java tests failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 3s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'java run' for 7.3.1.Final failed with exit code 1.
FAILED[java run][7.3.1.Final][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="7.3.1.Final" ./gradlew clean javaTest -Pcoordinates="org.jboss.narayana.jta:jta:5.12.4.Final"]
```

--- a/metadata/org.jboss.narayana.jta/jta/7.3.1.Final/reachability-metadata.json
+++ b/metadata/org.jboss.narayana.jta/jta/7.3.1.Final/reachability-metadata.json
@@ -1,0 +1,149 @@
+{
+  "reflection" : [
+    {
+      "condition" : {
+        "typeReached" : "com.arjuna.ats.internal.jta.resources.arjunacore.XAOnePhaseResource"
+      },
+      "type" : "byte[]",
+      "serializable" : true
+    },
+    {
+      "condition" : {
+        "typeReached" : "com.arjuna.ats.jta.xa.XidImple"
+      },
+      "type" : "byte[]",
+      "serializable" : true
+    },
+    {
+      "condition" : {
+        "typeReached" : "com.arjuna.ats.jta.common.jtaPropertyManager"
+      },
+      "type" : "ch.qos.logback.classic.Logger"
+    },
+    {
+      "condition" : {
+        "typeReached" : "com.arjuna.ats.internal.jta.resources.arjunacore.XAResourceRecord"
+      },
+      "type" : "com.arjuna.ats.arjuna.common.RecoveryEnvironmentBean",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "com.arjuna.ats.jta.common.jtaPropertyManager"
+      },
+      "type" : "com.arjuna.ats.jta.common.JTAEnvironmentBean",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "com.arjuna.ats.jta.logging.jtaLogger"
+      },
+      "type" : "com.arjuna.ats.jta.logging.jtaI18NLogger_$logger",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [
+            "org.jboss.logging.Logger"
+          ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "com.arjuna.ats.jta.logging.jtaLogger"
+      },
+      "type" : "com.arjuna.ats.jta.logging.jtaI18NLogger_$logger_en"
+    },
+    {
+      "condition" : {
+        "typeReached" : "com.arjuna.ats.jta.logging.jtaLogger"
+      },
+      "type" : "com.arjuna.ats.jta.logging.jtaI18NLogger_$logger_en_US"
+    },
+    {
+      "condition" : {
+        "typeReached" : "com.arjuna.ats.internal.jta.resources.arjunacore.XAOnePhaseResource"
+      },
+      "type" : "java.lang.String",
+      "serializable" : true
+    },
+    {
+      "condition" : {
+        "typeReached" : "com.arjuna.ats.jta.common.jtaPropertyManager"
+      },
+      "type" : "javax.xml.stream.XMLInputFactory"
+    },
+    {
+      "condition" : {
+        "typeReached" : "com.arjuna.ats.jta.common.jtaPropertyManager"
+      },
+      "type" : "org.apache.log4j.LogManager"
+    },
+    {
+      "condition" : {
+        "typeReached" : "com.arjuna.ats.jta.common.jtaPropertyManager"
+      },
+      "type" : "org.apache.logging.log4j.Logger"
+    },
+    {
+      "condition" : {
+        "typeReached" : "com.arjuna.ats.jta.common.jtaPropertyManager"
+      },
+      "type" : "org.jboss.logmanager.LogManager"
+    },
+    {
+      "condition" : {
+        "typeReached" : "com.arjuna.ats.internal.jta.resources.arjunacore.XAOnePhaseResource"
+      },
+      "type" : "sun.security.provider.SHA",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        }
+      ]
+    }
+  ],
+  "resources" : [
+    {
+      "condition" : {
+        "typeReached" : "com.arjuna.ats.jta.common.jtaPropertyManager"
+      },
+      "glob" : "ConfigurationInfo.properties"
+    },
+    {
+      "condition" : {
+        "typeReached" : "com.arjuna.ats.jta.common.jtaPropertyManager"
+      },
+      "glob" : "META-INF/services/java.time.zone.ZoneRulesProvider"
+    },
+    {
+      "condition" : {
+        "typeReached" : "com.arjuna.ats.jta.common.jtaPropertyManager"
+      },
+      "glob" : "META-INF/services/org.jboss.logging.LoggerProvider"
+    },
+    {
+      "condition" : {
+        "typeReached" : "com.arjuna.ats.jta.common.jtaPropertyManager"
+      },
+      "glob" : "default-jbossts-properties.xml"
+    },
+    {
+      "condition" : {
+        "typeReached" : "com.arjuna.ats.jta.common.jtaPropertyManager"
+      },
+      "glob" : "jbossts-properties.xml"
+    }
+  ]
+}

--- a/metadata/org.jboss.narayana.jta/jta/index.json
+++ b/metadata/org.jboss.narayana.jta/jta/index.json
@@ -1,6 +1,21 @@
 [
   {
     "latest" : true,
+    "metadata-version" : "7.3.1.Final",
+    "tested-versions" : [
+      "7.3.1.Final"
+    ],
+    "allowed-packages" : [
+      "org.jboss.narayana.jta",
+      "com.arjuna.ats.arjuna.common",
+      "com.arjuna.ats.internal.arjuna.utils",
+      "com.arjuna.ats.arjuna.logging",
+      "com.arjuna.ats.jta.common",
+      "com.arjuna.ats.jta.logging",
+      "com.arjuna.common.logging"
+    ]
+  },
+  {
     "metadata-version" : "5.12.4.Final",
     "source-code-url" : "https://repo1.maven.org/maven2/org/jboss/narayana/jta/jta/$version$/jta-$version$-sources.jar",
     "repository-url" : "https://github.com/jbosstm/narayana",

--- a/metadata/org.jboss.narayana.jta/jta/index.json
+++ b/metadata/org.jboss.narayana.jta/jta/index.json
@@ -2,17 +2,17 @@
   {
     "latest" : true,
     "metadata-version" : "7.3.1.Final",
+    "source-code-url" : "https://repo1.maven.org/maven2/org/jboss/narayana/jta/jta/$version$/jta-$version$-sources.jar",
+    "repository-url" : "https://github.com/jbosstm/narayana",
+    "test-code-url" : "https://github.com/jbosstm/narayana/tree/$version$/ArjunaJTA/jta/tests/classes",
+    "documentation-url" : "https://repo1.maven.org/maven2/org/jboss/narayana/jta/jta/$version$/jta-$version$-javadoc.jar",
+    "description" : "Narayana JTA provides a Java Transaction API implementation built on the Narayana transaction manager. It coordinates transactional work and recovery across local resources, application servers, and XA-capable systems.",
     "tested-versions" : [
       "7.3.1.Final"
     ],
     "allowed-packages" : [
-      "org.jboss.narayana.jta",
-      "com.arjuna.ats.arjuna.common",
-      "com.arjuna.ats.internal.arjuna.utils",
-      "com.arjuna.ats.arjuna.logging",
-      "com.arjuna.ats.jta.common",
-      "com.arjuna.ats.jta.logging",
-      "com.arjuna.common.logging"
+      "com.arjuna.ats.internal.jta",
+      "com.arjuna.ats.jta"
     ]
   },
   {

--- a/stats/org.jboss.narayana.jta/jta/7.3.1.Final/execution-metrics.json
+++ b/stats/org.jboss.narayana.jta/jta/7.3.1.Final/execution-metrics.json
@@ -1,0 +1,100 @@
+{
+  "fix_java_run_fail:2026-05-02": {
+    "timestamp": "2026-05-02T18:00:25.765485Z",
+    "library": "org.jboss.narayana.jta:jta:7.3.1.Final",
+    "starting_commit": "163802c7ce2105976e513787d25229467ff1790c",
+    "ending_commit": "fe49ad8869445f3168b9b48e08153fc353701fd1",
+    "previous_library": "org.jboss.narayana.jta:jta:5.12.4.Final",
+    "strategy_name": "java_run_iterative_with_coverage_sources_pi_gpt-5.5",
+    "agent": "pi",
+    "model": "oca/gpt-5.5",
+    "status": "success",
+    "stats": {
+      "version": "7.3.1.Final",
+      "dynamicAccess": {
+        "breakdown": {
+          "reflection": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 9,
+            "totalCalls": 9
+          }
+        },
+        "coverageRatio": 1.0,
+        "coveredCalls": 9,
+        "totalCalls": 9
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 725,
+          "missed": 17316,
+          "ratio": 0.040186,
+          "total": 18041
+        },
+        "line": {
+          "covered": 223,
+          "missed": 4962,
+          "ratio": 0.043009,
+          "total": 5185
+        },
+        "method": {
+          "covered": 37,
+          "missed": 930,
+          "ratio": 0.038263,
+          "total": 967
+        }
+      }
+    },
+    "previous_library_stats": {
+      "version": "5.12.4.Final",
+      "dynamicAccess": {
+        "breakdown": {
+          "reflection": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 8,
+            "totalCalls": 8
+          }
+        },
+        "coverageRatio": 1.0,
+        "coveredCalls": 8,
+        "totalCalls": 8
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 759,
+          "missed": 18148,
+          "ratio": 0.040144,
+          "total": 18907
+        },
+        "line": {
+          "covered": 222,
+          "missed": 4903,
+          "ratio": 0.043317,
+          "total": 5125
+        },
+        "method": {
+          "covered": 38,
+          "missed": 919,
+          "ratio": 0.039707,
+          "total": 957
+        }
+      }
+    },
+    "metrics": {
+      "input_tokens_used": 117371,
+      "cached_input_tokens_used": 1968640,
+      "output_tokens_used": 8846,
+      "iterations": 1,
+      "cost_usd": 1.2497,
+      "tested_library_loc": 223,
+      "metadata_entries": 26,
+      "test_only_metadata_entries": 14,
+      "previous_library_metadata_entries": 24,
+      "code_coverage_percent": 4.3,
+      "previous_library_coverage_percent": 4.33
+    },
+    "artifacts": {
+      "test_file": "tests/src/org.jboss.narayana.jta/jta/7.3.1.Final/src/test/java/org_jboss_narayana_jta/jta/JNDIManagerTest.java",
+      "metadata_file": "metadata/org.jboss.narayana.jta/jta/7.3.1.Final/reachability-metadata.json"
+    }
+  }
+}

--- a/stats/org.jboss.narayana.jta/jta/7.3.1.Final/stats.json
+++ b/stats/org.jboss.narayana.jta/jta/7.3.1.Final/stats.json
@@ -1,0 +1,37 @@
+{
+  "versions" : [ {
+    "version" : "7.3.1.Final",
+    "dynamicAccess" : {
+      "breakdown" : {
+        "reflection" : {
+          "coverageRatio" : 1.0,
+          "coveredCalls" : 9,
+          "totalCalls" : 9
+        }
+      },
+      "coverageRatio" : 1.0,
+      "coveredCalls" : 9,
+      "totalCalls" : 9
+    },
+    "libraryCoverage" : {
+      "instruction" : {
+        "covered" : 725,
+        "missed" : 17316,
+        "ratio" : 0.040186,
+        "total" : 18041
+      },
+      "line" : {
+        "covered" : 223,
+        "missed" : 4962,
+        "ratio" : 0.043009,
+        "total" : 5185
+      },
+      "method" : {
+        "covered" : 37,
+        "missed" : 930,
+        "ratio" : 0.038263,
+        "total" : 967
+      }
+    }
+  } ]
+}

--- a/tests/src/org.jboss.narayana.jta/jta/7.3.1.Final/.gitignore
+++ b/tests/src/org.jboss.narayana.jta/jta/7.3.1.Final/.gitignore
@@ -1,0 +1,4 @@
+gradlew.bat
+gradlew
+gradle/
+build/

--- a/tests/src/org.jboss.narayana.jta/jta/7.3.1.Final/build.gradle
+++ b/tests/src/org.jboss.narayana.jta/jta/7.3.1.Final/build.gradle
@@ -1,0 +1,29 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+plugins {
+    id "org.graalvm.internal.tck"
+}
+
+String libraryVersion = tck.testedLibraryVersion.get()
+
+dependencies {
+    testImplementation "org.jboss.narayana.jta:jta:$libraryVersion"
+    testImplementation 'org.assertj:assertj-core:3.22.0'
+    testImplementation 'org.jboss.logging:jboss-logging:3.2.1.Final'
+    testImplementation 'org.jboss:jboss-transaction-spi:7.6.1.Final'
+}
+
+graalvmNative {
+    agent {
+        defaultMode = "conditional"
+        modes {
+            conditional {
+                userCodeFilterPath = "user-code-filter.json"
+            }
+        }
+    }
+}

--- a/tests/src/org.jboss.narayana.jta/jta/7.3.1.Final/gradle.properties
+++ b/tests/src/org.jboss.narayana.jta/jta/7.3.1.Final/gradle.properties
@@ -1,0 +1,6 @@
+# Optional explicit tested library coordinates (format: group:artifact:version).
+# If omitted, fallback resolution in org.graalvm.internal.tck.gradle still uses
+# environment variable GVM_TCK_LC and then this property.
+library.coordinates = org.jboss.narayana.jta:jta:7.3.1.Final
+library.version = 7.3.1.Final
+metadata.dir = org.jboss.narayana.jta/jta/7.3.1.Final/

--- a/tests/src/org.jboss.narayana.jta/jta/7.3.1.Final/settings.gradle
+++ b/tests/src/org.jboss.narayana.jta/jta/7.3.1.Final/settings.gradle
@@ -1,0 +1,20 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+pluginManagement {
+    def tckPath = Objects.requireNonNullElse(
+            System.getenv("GVM_TCK_TCKDIR"),
+            "../../../../tck-build-logic"
+    )
+    includeBuild(tckPath)
+    repositories { mavenLocal(); gradlePluginPortal() }
+}
+
+plugins {
+    id "org.graalvm.internal.tck-settings" version "1.0.0-SNAPSHOT"
+}
+
+rootProject.name = 'org.jboss.narayana.jta.jta_tests'

--- a/tests/src/org.jboss.narayana.jta/jta/7.3.1.Final/src/test/java/com/arjuna/common/logging/commonI18NLogger_$logger.java
+++ b/tests/src/org.jboss.narayana.jta/jta/7.3.1.Final/src/test/java/com/arjuna/common/logging/commonI18NLogger_$logger.java
@@ -1,0 +1,75 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package com.arjuna.common.logging;
+
+import java.io.Serializable;
+import java.net.URL;
+import java.text.MessageFormat;
+import java.util.Locale;
+
+import org.jboss.logging.Logger;
+
+// CheckStyle: start generated
+public class commonI18NLogger_$logger implements commonI18NLogger, Serializable {
+    private static final long serialVersionUID = 1L;
+
+    private final Logger log;
+
+    public commonI18NLogger_$logger(Logger log) {
+        this.log = log;
+    }
+
+    @Override
+    public void warn_could_not_find_manifest(String manifestName, Throwable cause) {
+        log.warn(format("Could not find manifest {0}", manifestName), cause);
+    }
+
+    @Override
+    public void warn_could_not_find_config_file(URL url) {
+        log.warn(format("Could not find configuration file, URL was: {0}", url));
+    }
+
+    @Override
+    public void warn_common_ClassloadingUtility_1() {
+        log.warn("className is null");
+    }
+
+    @Override
+    public void warn_common_ClassloadingUtility_2(String className, Throwable cause) {
+        log.warn(format("attempt to load {0} threw ClassNotFound. Wrong classloader?", className), cause);
+    }
+
+    @Override
+    public void warn_common_ClassloadingUtility_3(String className, String interfaceName, Throwable cause) {
+        log.warn(format("class {0} does not implement {1}", className, interfaceName), cause);
+    }
+
+    @Override
+    public String warn_common_ClassloadingUtility_4(String className, Throwable cause) {
+        return format("cannot create new instance of {0}", className);
+    }
+
+    @Override
+    public String warn_common_ClassloadingUtility_5(String className, Throwable cause) {
+        return format("cannot access {0}", className);
+    }
+
+    @Override
+    public void warn_common_ClassloadingUtility_6(String value, Throwable cause) {
+        log.warn(format("cannot initialize from string {0}", value), cause);
+    }
+
+    @Override
+    public String warn_common_NoSuchMethodException(String className, Throwable cause) {
+        return format("cannot create new instance of {0} as no zero-argument constructor exists", className);
+    }
+
+    private static String format(String pattern, Object... arguments) {
+        return new MessageFormat(pattern, Locale.ROOT).format(arguments);
+    }
+}
+// CheckStyle: stop generated

--- a/tests/src/org.jboss.narayana.jta/jta/7.3.1.Final/src/test/java/org_jboss_narayana_jta/jta/JNDIManagerTest.java
+++ b/tests/src/org.jboss.narayana.jta/jta/7.3.1.Final/src/test/java/org_jboss_narayana_jta/jta/JNDIManagerTest.java
@@ -1,0 +1,114 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package org_jboss_narayana_jta.jta;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import javax.naming.InitialContext;
+import javax.naming.NamingException;
+import javax.transaction.Status;
+import javax.transaction.Synchronization;
+import javax.transaction.TransactionSynchronizationRegistry;
+
+import com.arjuna.ats.jta.common.JTAEnvironmentBean;
+import com.arjuna.ats.jta.common.jtaPropertyManager;
+import com.arjuna.ats.jta.utils.JNDIManager;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class JNDIManagerTest {
+    @Test
+    void bindTransactionSynchronizationRegistryImplementationUsesConfiguredClass() throws Exception {
+        JTAEnvironmentBean environmentBean = jtaPropertyManager.getJTAEnvironmentBean();
+        String originalClassName = environmentBean.getTransactionSynchronizationRegistryClassName();
+        String originalJndiName = environmentBean.getTransactionSynchronizationRegistryJNDIContext();
+
+        try {
+            TestTransactionSynchronizationRegistry.CONSTRUCTOR_CALLS.set(0);
+            environmentBean.setTransactionSynchronizationRegistryClassName(
+                TestTransactionSynchronizationRegistry.class.getName()
+            );
+            environmentBean.setTransactionSynchronizationRegistryJNDIContext(
+                "java:/test/TransactionSynchronizationRegistry"
+            );
+
+            RecordingInitialContext initialContext = new RecordingInitialContext();
+
+            JNDIManager.bindJTATransactionSynchronizationRegistryImplementation(initialContext);
+
+            assertThat(TestTransactionSynchronizationRegistry.CONSTRUCTOR_CALLS).hasValue(1);
+            assertThat(initialContext.boundName).isEqualTo("java:/test/TransactionSynchronizationRegistry");
+            assertThat(initialContext.boundValue).isInstanceOf(TestTransactionSynchronizationRegistry.class);
+        } finally {
+            environmentBean.setTransactionSynchronizationRegistryClassName(originalClassName);
+            environmentBean.setTransactionSynchronizationRegistryJNDIContext(originalJndiName);
+        }
+    }
+
+    public static final class TestTransactionSynchronizationRegistry implements TransactionSynchronizationRegistry {
+        private static final AtomicInteger CONSTRUCTOR_CALLS = new AtomicInteger();
+
+        private final Map<Object, Object> resources = new HashMap<>();
+        private boolean rollbackOnly;
+
+        public TestTransactionSynchronizationRegistry() {
+            CONSTRUCTOR_CALLS.incrementAndGet();
+        }
+
+        @Override
+        public Object getTransactionKey() {
+            return this;
+        }
+
+        @Override
+        public void putResource(Object key, Object value) {
+            resources.put(key, value);
+        }
+
+        @Override
+        public Object getResource(Object key) {
+            return resources.get(key);
+        }
+
+        @Override
+        public void registerInterposedSynchronization(Synchronization synchronization) {
+        }
+
+        @Override
+        public int getTransactionStatus() {
+            return Status.STATUS_NO_TRANSACTION;
+        }
+
+        @Override
+        public void setRollbackOnly() {
+            rollbackOnly = true;
+        }
+
+        @Override
+        public boolean getRollbackOnly() {
+            return rollbackOnly;
+        }
+    }
+
+    private static final class RecordingInitialContext extends InitialContext {
+        private String boundName;
+        private Object boundValue;
+
+        private RecordingInitialContext() throws NamingException {
+            super(true);
+        }
+
+        @Override
+        public void rebind(String name, Object obj) {
+            boundName = name;
+            boundValue = obj;
+        }
+    }
+}

--- a/tests/src/org.jboss.narayana.jta/jta/7.3.1.Final/src/test/java/org_jboss_narayana_jta/jta/XAOnePhaseResourceTest.java
+++ b/tests/src/org.jboss.narayana.jta/jta/7.3.1.Final/src/test/java/org_jboss_narayana_jta/jta/XAOnePhaseResourceTest.java
@@ -1,0 +1,135 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package org_jboss_narayana_jta.jta;
+
+import java.io.Serializable;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import javax.transaction.xa.XAResource;
+import javax.transaction.xa.Xid;
+
+import com.arjuna.ats.arjuna.coordinator.TwoPhaseOutcome;
+import com.arjuna.ats.arjuna.state.InputObjectState;
+import com.arjuna.ats.arjuna.state.OutputObjectState;
+import com.arjuna.ats.internal.jta.resources.arjunacore.XAOnePhaseResource;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class XAOnePhaseResourceTest {
+    @Test
+    void packAndUnpackSerializableXaResource() throws Exception {
+        SerializableXaResource.COMMIT_CALLS.set(0);
+
+        XAOnePhaseResource resource = new XAOnePhaseResource(
+            new SerializableXaResource("pack-unpack"),
+            new SerializableXid(41, new byte[]{1, 2, 3}, new byte[]{4, 5, 6}),
+            null
+        );
+        OutputObjectState outputState = new OutputObjectState();
+
+        resource.pack(outputState);
+
+        XAOnePhaseResource restoredResource = new XAOnePhaseResource();
+        restoredResource.unpack(new InputObjectState(outputState));
+
+        assertThat(restoredResource.toString()).contains("pack-unpack");
+        assertThat(restoredResource.commit()).isEqualTo(TwoPhaseOutcome.FINISH_OK);
+        assertThat(SerializableXaResource.COMMIT_CALLS).hasValue(1);
+    }
+
+    private static final class SerializableXaResource implements XAResource, Serializable {
+        private static final long serialVersionUID = 1L;
+        private static final AtomicInteger COMMIT_CALLS = new AtomicInteger();
+
+        private final String name;
+
+        private SerializableXaResource(String name) {
+            this.name = name;
+        }
+
+        @Override
+        public void commit(Xid xid, boolean onePhase) {
+            COMMIT_CALLS.incrementAndGet();
+        }
+
+        @Override
+        public void end(Xid xid, int flags) {
+        }
+
+        @Override
+        public void forget(Xid xid) {
+        }
+
+        @Override
+        public int getTransactionTimeout() {
+            return 0;
+        }
+
+        @Override
+        public boolean isSameRM(XAResource xaResource) {
+            return xaResource == this;
+        }
+
+        @Override
+        public int prepare(Xid xid) {
+            return XA_OK;
+        }
+
+        @Override
+        public Xid[] recover(int flag) {
+            return new Xid[0];
+        }
+
+        @Override
+        public void rollback(Xid xid) {
+        }
+
+        @Override
+        public boolean setTransactionTimeout(int seconds) {
+            return true;
+        }
+
+        @Override
+        public void start(Xid xid, int flags) {
+        }
+
+        @Override
+        public String toString() {
+            return "SerializableXaResource[" + name + "]";
+        }
+    }
+
+    private static final class SerializableXid implements Xid, Serializable {
+        private static final long serialVersionUID = 1L;
+
+        private final int formatId;
+        private final byte[] globalTransactionId;
+        private final byte[] branchQualifier;
+
+        private SerializableXid(int formatId, byte[] globalTransactionId, byte[] branchQualifier) {
+            this.formatId = formatId;
+            this.globalTransactionId = globalTransactionId;
+            this.branchQualifier = branchQualifier;
+        }
+
+        @Override
+        public int getFormatId() {
+            return formatId;
+        }
+
+        @Override
+        public byte[] getGlobalTransactionId() {
+            return globalTransactionId;
+        }
+
+        @Override
+        public byte[] getBranchQualifier() {
+            return branchQualifier;
+        }
+    }
+}

--- a/tests/src/org.jboss.narayana.jta/jta/7.3.1.Final/src/test/java/org_jboss_narayana_jta/jta/XAResourceRecordTest.java
+++ b/tests/src/org.jboss.narayana.jta/jta/7.3.1.Final/src/test/java/org_jboss_narayana_jta/jta/XAResourceRecordTest.java
@@ -1,0 +1,152 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package org_jboss_narayana_jta.jta;
+
+import java.io.Serializable;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import javax.transaction.xa.XAResource;
+import javax.transaction.xa.Xid;
+
+import com.arjuna.ats.arjuna.ObjectType;
+import com.arjuna.ats.arjuna.state.InputObjectState;
+import com.arjuna.ats.arjuna.state.OutputObjectState;
+import com.arjuna.ats.internal.jta.resources.arjunacore.XAResourceRecord;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class XAResourceRecordTest {
+    @Test
+    void saveAndRestoreStateWithSerializableXaResource() throws Exception {
+        SerializableXaResource.COMMIT_CALLS.set(0);
+
+        XAResourceRecord resourceRecord = new XAResourceRecord(
+            null,
+            new SerializableXaResource("xa-resource-record"),
+            new SerializableXid(73, new byte[]{1, 3, 5}, new byte[]{2, 4, 6}),
+            null
+        );
+        resourceRecord.setProductName("Test Product");
+        resourceRecord.setProductVersion("2026.04");
+        resourceRecord.setJndiName("java:/test/XAResourceRecord");
+
+        OutputObjectState outputState = new OutputObjectState();
+
+        assertThat(resourceRecord.save_state(outputState, ObjectType.ANDPERSISTENT)).isTrue();
+
+        XAResourceRecord restoredRecord = new XAResourceRecord();
+
+        assertThat(restoredRecord.restore_state(new InputObjectState(outputState), ObjectType.ANDPERSISTENT)).isTrue();
+        assertThat(restoredRecord.getProductName()).isEqualTo("Test Product");
+        assertThat(restoredRecord.getProductVersion()).isEqualTo("2026.04");
+        assertThat(restoredRecord.getJndiName()).isEqualTo("java:/test/XAResourceRecord");
+        assertThat(restoredRecord.getXid().getFormatId()).isEqualTo(73);
+        assertThat(restoredRecord.getXid().getGlobalTransactionId()).containsExactly((byte) 1, (byte) 3, (byte) 5);
+        assertThat(restoredRecord.getXid().getBranchQualifier()).containsExactly((byte) 2, (byte) 4, (byte) 6);
+        assertThat(restoredRecord.toString()).contains("xa-resource-record");
+
+        XAResource restoredResource = (XAResource) restoredRecord.value();
+
+        assertThat(restoredResource).isInstanceOf(SerializableXaResource.class);
+
+        restoredResource.commit(restoredRecord.getXid(), true);
+
+        assertThat(SerializableXaResource.COMMIT_CALLS).hasValue(1);
+    }
+
+    private static final class SerializableXaResource implements XAResource, Serializable {
+        private static final long serialVersionUID = 1L;
+        private static final AtomicInteger COMMIT_CALLS = new AtomicInteger();
+
+        private final String name;
+
+        private SerializableXaResource(String name) {
+            this.name = name;
+        }
+
+        @Override
+        public void commit(Xid xid, boolean onePhase) {
+            COMMIT_CALLS.incrementAndGet();
+        }
+
+        @Override
+        public void end(Xid xid, int flags) {
+        }
+
+        @Override
+        public void forget(Xid xid) {
+        }
+
+        @Override
+        public int getTransactionTimeout() {
+            return 0;
+        }
+
+        @Override
+        public boolean isSameRM(XAResource xaResource) {
+            return xaResource == this;
+        }
+
+        @Override
+        public int prepare(Xid xid) {
+            return XA_OK;
+        }
+
+        @Override
+        public Xid[] recover(int flag) {
+            return new Xid[0];
+        }
+
+        @Override
+        public void rollback(Xid xid) {
+        }
+
+        @Override
+        public boolean setTransactionTimeout(int seconds) {
+            return true;
+        }
+
+        @Override
+        public void start(Xid xid, int flags) {
+        }
+
+        @Override
+        public String toString() {
+            return "SerializableXaResource[" + name + "]";
+        }
+    }
+
+    private static final class SerializableXid implements Xid, Serializable {
+        private static final long serialVersionUID = 1L;
+
+        private final int formatId;
+        private final byte[] globalTransactionId;
+        private final byte[] branchQualifier;
+
+        private SerializableXid(int formatId, byte[] globalTransactionId, byte[] branchQualifier) {
+            this.formatId = formatId;
+            this.globalTransactionId = globalTransactionId;
+            this.branchQualifier = branchQualifier;
+        }
+
+        @Override
+        public int getFormatId() {
+            return formatId;
+        }
+
+        @Override
+        public byte[] getGlobalTransactionId() {
+            return globalTransactionId;
+        }
+
+        @Override
+        public byte[] getBranchQualifier() {
+            return branchQualifier;
+        }
+    }
+}

--- a/tests/src/org.jboss.narayana.jta/jta/7.3.1.Final/src/test/resources/META-INF/native-image/org.jboss.narayana.jta.jta.tests/reflect-config.json
+++ b/tests/src/org.jboss.narayana.jta/jta/7.3.1.Final/src/test/resources/META-INF/native-image/org.jboss.narayana.jta.jta.tests/reflect-config.json
@@ -1,0 +1,83 @@
+[
+  {
+    "name": "com.arjuna.ats.arjuna.common.CoordinatorEnvironmentBean",
+    "methods": [
+      {
+        "name": "<init>",
+        "parameterTypes": []
+      }
+    ]
+  },
+  {
+    "name": "com.arjuna.ats.arjuna.common.CoreEnvironmentBean",
+    "methods": [
+      {
+        "name": "<init>",
+        "parameterTypes": []
+      }
+    ]
+  },
+  {
+    "name": "com.arjuna.ats.arjuna.common.RecoveryEnvironmentBean",
+    "methods": [
+      {
+        "name": "<init>",
+        "parameterTypes": []
+      }
+    ]
+  },
+  {
+    "name": "com.arjuna.ats.arjuna.logging.arjunaI18NLogger_$logger",
+    "methods": [
+      {
+        "name": "<init>",
+        "parameterTypes": ["org.jboss.logging.Logger"]
+      }
+    ]
+  },
+  {
+    "name": "com.arjuna.ats.internal.arjuna.utils.SocketProcessId",
+    "methods": [
+      {
+        "name": "<init>",
+        "parameterTypes": []
+      }
+    ]
+  },
+  {
+    "name": "com.arjuna.ats.jta.common.JTAEnvironmentBean",
+    "methods": [
+      {
+        "name": "<init>",
+        "parameterTypes": []
+      }
+    ]
+  },
+  {
+    "name": "com.arjuna.ats.jta.logging.jtaI18NLogger_$logger",
+    "methods": [
+      {
+        "name": "<init>",
+        "parameterTypes": ["org.jboss.logging.Logger"]
+      }
+    ]
+  },
+  {
+    "name": "com.arjuna.common.logging.commonI18NLogger_$logger",
+    "methods": [
+      {
+        "name": "<init>",
+        "parameterTypes": ["org.jboss.logging.Logger"]
+      }
+    ]
+  },
+  {
+    "name": "org_jboss_narayana_jta.jta.JNDIManagerTest$TestTransactionSynchronizationRegistry",
+    "methods": [
+      {
+        "name": "<init>",
+        "parameterTypes": []
+      }
+    ]
+  }
+]

--- a/tests/src/org.jboss.narayana.jta/jta/7.3.1.Final/src/test/resources/META-INF/native-image/org.jboss.narayana.jta.jta.tests/serialization-config.json
+++ b/tests/src/org.jboss.narayana.jta/jta/7.3.1.Final/src/test/resources/META-INF/native-image/org.jboss.narayana.jta.jta.tests/serialization-config.json
@@ -1,0 +1,17 @@
+[
+  {
+    "name": "java.lang.String"
+  },
+  {
+    "name": "org_jboss_narayana_jta.jta.XAOnePhaseResourceTest$SerializableXaResource"
+  },
+  {
+    "name": "org_jboss_narayana_jta.jta.XAOnePhaseResourceTest$SerializableXid"
+  },
+  {
+    "name": "org_jboss_narayana_jta.jta.XAResourceRecordTest$SerializableXaResource"
+  },
+  {
+    "name": "org_jboss_narayana_jta.jta.XAResourceRecordTest$SerializableXid"
+  }
+]

--- a/tests/src/org.jboss.narayana.jta/jta/7.3.1.Final/src/test/resources/META-INF/native-image/reachability-metadata.json
+++ b/tests/src/org.jboss.narayana.jta/jta/7.3.1.Final/src/test/resources/META-INF/native-image/reachability-metadata.json
@@ -1,0 +1,70 @@
+{
+  "reflection" : [
+    {
+      "condition" : {
+        "typeReached" : "com.arjuna.ats.jta.common.jtaPropertyManager"
+      },
+      "type" : "com.arjuna.common.logging.commonI18NLogger_$logger",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [
+            "org.jboss.logging.Logger"
+          ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "com.arjuna.ats.jta.common.jtaPropertyManager"
+      },
+      "type" : "com.arjuna.common.logging.commonI18NLogger_$logger_en"
+    },
+    {
+      "condition" : {
+        "typeReached" : "com.arjuna.ats.jta.common.jtaPropertyManager"
+      },
+      "type" : "com.arjuna.common.logging.commonI18NLogger_$logger_en_US"
+    },
+    {
+      "condition" : {
+        "typeReached" : "com.arjuna.ats.jta.utils.JNDIManager"
+      },
+      "type" : "org_jboss_narayana_jta.jta.JNDIManagerTest$TestTransactionSynchronizationRegistry",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "com.arjuna.ats.internal.jta.resources.arjunacore.XAOnePhaseResource"
+      },
+      "type" : "org_jboss_narayana_jta.jta.XAOnePhaseResourceTest$SerializableXaResource",
+      "serializable" : true
+    },
+    {
+      "condition" : {
+        "typeReached" : "com.arjuna.ats.internal.jta.resources.arjunacore.XAOnePhaseResource"
+      },
+      "type" : "org_jboss_narayana_jta.jta.XAOnePhaseResourceTest$SerializableXid",
+      "serializable" : true
+    },
+    {
+      "condition" : {
+        "typeReached" : "com.arjuna.ats.internal.jta.resources.arjunacore.XAResourceRecord"
+      },
+      "type" : "org_jboss_narayana_jta.jta.XAResourceRecordTest$SerializableXaResource",
+      "serializable" : true
+    },
+    {
+      "condition" : {
+        "typeReached" : "com.arjuna.ats.internal.jta.resources.arjunacore.XAResourceRecord"
+      },
+      "type" : "org_jboss_narayana_jta.jta.XAResourceRecordTest$SerializableXid",
+      "serializable" : true
+    }
+  ]
+}

--- a/tests/src/org.jboss.narayana.jta/jta/7.3.1.Final/test-results-native/test/TEST-junit-jupiter.xml
+++ b/tests/src/org.jboss.narayana.jta/jta/7.3.1.Final/test-results-native/test/TEST-junit-jupiter.xml
@@ -1,0 +1,76 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<testsuite name="JUnit Jupiter" tests="3" skipped="0" failures="0" errors="0" time="0.002" hostname="kung-fu-workstation" timestamp="2026-05-02T19:58:19">
+<properties>
+<property name="file.encoding" value="UTF-8"/>
+<property name="file.separator" value="/"/>
+<property name="java.class.path" value=""/>
+<property name="java.class.version" value="69.0"/>
+<property name="java.endorsed.dirs" value=""/>
+<property name="java.ext.dirs" value=""/>
+<property name="java.io.tmpdir" value="/tmp"/>
+<property name="java.library.path" value="/usr/lib64:/lib64:/lib:/usr/lib"/>
+<property name="java.runtime.name" value="GraalVM Runtime Environment"/>
+<property name="java.runtime.version" value="25.0.3+9-LTS-jvmci-b01"/>
+<property name="java.specification.name" value="Java Platform API Specification"/>
+<property name="java.specification.vendor" value="Oracle Corporation"/>
+<property name="java.specification.version" value="25"/>
+<property name="java.vendor" value="Oracle Corporation"/>
+<property name="java.vendor.url" value="https://www.graalvm.org/"/>
+<property name="java.vendor.version" value="Oracle GraalVM 25.0.3+9.1"/>
+<property name="java.version" value="25.0.3"/>
+<property name="java.version.date" value="2026-04-21"/>
+<property name="java.vm.info" value="serial gc, compressed references"/>
+<property name="java.vm.name" value="Substrate VM"/>
+<property name="java.vm.specification.name" value="Java Virtual Machine Specification"/>
+<property name="java.vm.specification.vendor" value="Oracle Corporation"/>
+<property name="java.vm.specification.version" value="25"/>
+<property name="java.vm.vendor" value="Oracle Corporation"/>
+<property name="java.vm.version" value="25.0.3+9-LTS"/>
+<property name="jdk.lang.Process.launchMechanism" value="FORK"/>
+<property name="junit.jupiter.execution.timeout.default" value="60 s"/>
+<property name="junit.jupiter.execution.timeout.mode" value="enabled"/>
+<property name="junit.jupiter.execution.timeout.thread.mode.default" value="separate_thread"/>
+<property name="line.separator" value="
+"/>
+<property name="native.encoding" value="UTF-8"/>
+<property name="org.graalvm.nativeimage.imagecode" value="runtime"/>
+<property name="org.graalvm.nativeimage.kind" value="executable"/>
+<property name="os.arch" value="amd64"/>
+<property name="os.name" value="Linux"/>
+<property name="os.version" value="6.17.0-22-generic"/>
+<property name="path.separator" value=":"/>
+<property name="stderr.encoding" value="UTF-8"/>
+<property name="stdin.encoding" value="UTF-8"/>
+<property name="stdout.encoding" value="UTF-8"/>
+<property name="sun.arch.data.model" value="64"/>
+<property name="sun.jnu.encoding" value="UTF-8"/>
+<property name="user.country" value="US"/>
+<property name="user.dir" value="/home/vjovanov/c/forge/forge5/graalvm-reachability-metadata/forge/local_repositories/forge_worktrees/1844-4c348ee5/tests/src/org.jboss.narayana.jta/jta/7.3.1.Final"/>
+<property name="user.home" value="/home/vjovanov"/>
+<property name="user.language" value="en"/>
+<property name="user.name" value="vjovanov"/>
+<property name="user.timezone" value="Europe/Zurich"/>
+</properties>
+<testcase name="packAndUnpackSerializableXaResource()" classname="org_jboss_narayana_jta.jta.XAOnePhaseResourceTest" time="0">
+<system-out><![CDATA[
+unique-id: [engine:junit-jupiter]/[class:org_jboss_narayana_jta.jta.XAOnePhaseResourceTest]/[method:packAndUnpackSerializableXaResource()]
+display-name: packAndUnpackSerializableXaResource()
+]]></system-out>
+</testcase>
+<testcase name="bindTransactionSynchronizationRegistryImplementationUsesConfiguredClass()" classname="org_jboss_narayana_jta.jta.JNDIManagerTest" time="0.001">
+<system-out><![CDATA[
+unique-id: [engine:junit-jupiter]/[class:org_jboss_narayana_jta.jta.JNDIManagerTest]/[method:bindTransactionSynchronizationRegistryImplementationUsesConfiguredClass()]
+display-name: bindTransactionSynchronizationRegistryImplementationUsesConfiguredClass()
+]]></system-out>
+</testcase>
+<testcase name="saveAndRestoreStateWithSerializableXaResource()" classname="org_jboss_narayana_jta.jta.XAResourceRecordTest" time="0">
+<system-out><![CDATA[
+unique-id: [engine:junit-jupiter]/[class:org_jboss_narayana_jta.jta.XAResourceRecordTest]/[method:saveAndRestoreStateWithSerializableXaResource()]
+display-name: saveAndRestoreStateWithSerializableXaResource()
+]]></system-out>
+</testcase>
+<system-out><![CDATA[
+unique-id: [engine:junit-jupiter]
+display-name: JUnit Jupiter
+]]></system-out>
+</testsuite>

--- a/tests/src/org.jboss.narayana.jta/jta/7.3.1.Final/test-results-native/test/TEST-junit-vintage.xml
+++ b/tests/src/org.jboss.narayana.jta/jta/7.3.1.Final/test-results-native/test/TEST-junit-vintage.xml
@@ -1,0 +1,58 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<testsuite name="JUnit Vintage" tests="0" skipped="0" failures="0" errors="0" time="0" hostname="kung-fu-workstation" timestamp="2026-05-02T19:58:19">
+<properties>
+<property name="file.encoding" value="UTF-8"/>
+<property name="file.separator" value="/"/>
+<property name="java.class.path" value=""/>
+<property name="java.class.version" value="69.0"/>
+<property name="java.endorsed.dirs" value=""/>
+<property name="java.ext.dirs" value=""/>
+<property name="java.io.tmpdir" value="/tmp"/>
+<property name="java.library.path" value="/usr/lib64:/lib64:/lib:/usr/lib"/>
+<property name="java.runtime.name" value="GraalVM Runtime Environment"/>
+<property name="java.runtime.version" value="25.0.3+9-LTS-jvmci-b01"/>
+<property name="java.specification.name" value="Java Platform API Specification"/>
+<property name="java.specification.vendor" value="Oracle Corporation"/>
+<property name="java.specification.version" value="25"/>
+<property name="java.vendor" value="Oracle Corporation"/>
+<property name="java.vendor.url" value="https://www.graalvm.org/"/>
+<property name="java.vendor.version" value="Oracle GraalVM 25.0.3+9.1"/>
+<property name="java.version" value="25.0.3"/>
+<property name="java.version.date" value="2026-04-21"/>
+<property name="java.vm.info" value="serial gc, compressed references"/>
+<property name="java.vm.name" value="Substrate VM"/>
+<property name="java.vm.specification.name" value="Java Virtual Machine Specification"/>
+<property name="java.vm.specification.vendor" value="Oracle Corporation"/>
+<property name="java.vm.specification.version" value="25"/>
+<property name="java.vm.vendor" value="Oracle Corporation"/>
+<property name="java.vm.version" value="25.0.3+9-LTS"/>
+<property name="jdk.lang.Process.launchMechanism" value="FORK"/>
+<property name="junit.jupiter.execution.timeout.default" value="60 s"/>
+<property name="junit.jupiter.execution.timeout.mode" value="enabled"/>
+<property name="junit.jupiter.execution.timeout.thread.mode.default" value="separate_thread"/>
+<property name="line.separator" value="
+"/>
+<property name="native.encoding" value="UTF-8"/>
+<property name="org.graalvm.nativeimage.imagecode" value="runtime"/>
+<property name="org.graalvm.nativeimage.kind" value="executable"/>
+<property name="os.arch" value="amd64"/>
+<property name="os.name" value="Linux"/>
+<property name="os.version" value="6.17.0-22-generic"/>
+<property name="path.separator" value=":"/>
+<property name="stderr.encoding" value="UTF-8"/>
+<property name="stdin.encoding" value="UTF-8"/>
+<property name="stdout.encoding" value="UTF-8"/>
+<property name="sun.arch.data.model" value="64"/>
+<property name="sun.jnu.encoding" value="UTF-8"/>
+<property name="user.country" value="US"/>
+<property name="user.dir" value="/home/vjovanov/c/forge/forge5/graalvm-reachability-metadata/forge/local_repositories/forge_worktrees/1844-4c348ee5/tests/src/org.jboss.narayana.jta/jta/7.3.1.Final"/>
+<property name="user.home" value="/home/vjovanov"/>
+<property name="user.language" value="en"/>
+<property name="user.name" value="vjovanov"/>
+<property name="user.timezone" value="Europe/Zurich"/>
+</properties>
+<system-out><![CDATA[
+unique-id: [engine:junit-vintage]
+display-name: JUnit Vintage
+]]></system-out>
+</testsuite>

--- a/tests/src/org.jboss.narayana.jta/jta/7.3.1.Final/user-code-filter.json
+++ b/tests/src/org.jboss.narayana.jta/jta/7.3.1.Final/user-code-filter.json
@@ -1,0 +1,10 @@
+{
+  "rules" : [
+    {
+      "excludeClasses" : "**"
+    },
+    {
+      "includeClasses" : "org.jboss.narayana.jta.**"
+    }
+  ]
+}

--- a/tests/src/org.jboss.narayana.jta/jta/7.3.1.Final/user-code-filter.json
+++ b/tests/src/org.jboss.narayana.jta/jta/7.3.1.Final/user-code-filter.json
@@ -4,7 +4,10 @@
       "excludeClasses" : "**"
     },
     {
-      "includeClasses" : "org.jboss.narayana.jta.**"
+      "includeClasses" : "com.arjuna.ats.internal.jta.**"
+    },
+    {
+      "includeClasses" : "com.arjuna.ats.jta.**"
     }
   ]
 }

